R/utils.R

```diff
@@ -8,14 +8,39 @@
 #' @usage lhs \%>\% rhs
 NULL
 
-at_selection <- function(nm, .at){
-  if (is_quosures(.at)) {
-    lifecycle::deprecate_warn("0.4.0", I("using `vars()` in .at"))
+at_selection <- function(x, at, error_arg = caller_arg(at), error_call = caller_env()) {
+  if (is_formula(at)) {
+    at <- rlang::as_function(at, arg = error_arg, call = error_call)
+  }
+  if (is.function(at)) {
+    at <- at(names(x))
+  }
 
+  if (is.numeric(at) || is.logical(at) ||is.character(at)) {
+    if (is.character(at) && !is.null(names(x))) {
+      at <- intersect(at, names(x))
+    }
+
+    at <- vec_as_subscript(at, arg = "at", call = error_call)
+    vec_as_location(
+      at,
+      length(x),
+      names(x),
```

Comment:
I wonder if you want to use `names2()` everywhere to avoid this error?

```r
keep_at(1, letters)
#> Error in `vec_as_location()` at purrr/R/utils.R:25:4:
#> ! Can't use character names to index an unnamed vector.
```

Reply:
A second example you could use as a test:

```r
keep_at(1, \(nms) nms %in% letters)
#> Error in `keep_at()`:
#> ! Must subset elements with a valid subscript vector.
#> ℹ Logical subscripts must match the size of the indexed input.
#> ✖ Input has size 1 but subscript `at` has size 0.
```

This one tests the function case